apiClientDotNet/Listeners/ConnectionListener.cs

```diff
@@ -1,13 +1,10 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
-using apiClientDotNet.Models;
+﻿using apiClientDotNet.Models;
 
 namespace apiClientDotNet.Listeners
 {
-    public abstract class ConnectionListener
+    public interface ConnectionListener
```

Comment:
Unfortunatley with this customer code will fail to compile. Just because there is a slight difference between implementing interface method and overriding virtual method:
        public override void OnCreate()
        {
            
        }
        
        public void OnCreate()
        {
            
        }

We can keep the classes, but mark as deprecated, and create interfaces with correct naming like IConnectionListener.